packages/@react-aria/datepicker/src/useDateSegment.ts

```diff
@@ -346,7 +346,7 @@ export function useDateSegment(segment: DateSegment, state: DateFieldState, ref:
   // This is needed because VoiceOver on iOS does not announce groups.
   let name = segment.type === 'literal' ? '' : displayNames.of(segment.type);
   let labelProps = useLabels({
-    'aria-label': (ariaLabel ? ariaLabel + ' ' : '') + name,
+    'aria-label': `${name}${ariaLabel ? `, ${ariaLabel}` : ''}${ariaLabelledBy ? ', ' : ''}`,
```

Comment:
commas are being added to get the screen reader to make a small pause?

Reply:
Yes. Particularly in DateRangePicker, the labels become a  something of a string of words. For example, "5 - May, Date range Start date month, stepper." With the changes suggested in this PR, the accessibility name will read from the value most likely to change to the value least likely to change as the user navigates between segments. "5 - May, month, Start date, Date range, stepper" The commas help make more sense of the reordered string.